openmc/data/function.py

```diff
@@ -432,6 +488,63 @@ def functions(self, functions):
         self._functions = functions
 
 
+class Piecewise(EqualityMixin):
```

Comment:
In GND terminology, this would be a `Regions1D`. Perhaps we should consider adopting that naming scheme? See [here](https://www.oecd-nea.org/science/wpec/sg38/Meetings/2015_May/generalDataContainers.pdf) for a draft spec of data containers (note that it is more than a year out of date).